### ISSUE TITLE
[Suggestion] Ship QOL Targeting Improvements

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Starship-EVO#3447]
After playing Star Wars Squadrons for a bit, there are some combat targeting features in that game that may work very well in SEVO.

1.	Target Location

This one is simple enough, when you are targeting a ship that isn’t in your field of view, an arrow points in the direction of the targeted ship. Additionally, a better way to distinguish the ship you are targeting would be very useful. Perhaps the target is more heavily bolded and/or has colored brackets.
![image](https://user-images.githubusercontent.com/36980446/103850045-877c8080-5063-11eb-87db-6bb795cf433d.png)
![image](https://user-images.githubusercontent.com/36980446/103850299-2608e180-5064-11eb-9c34-d6f0a87c03a3.png)
Hard to tell which one is targeted, especially if you are in a dense crowd of ships. Oh, and welcome to one of my many ship graveyards.

2.	Target Selection

The feature where you can target a specific portion of the ship is kind of useless since it is impossible to tell exactly where vital systems are, difficult to click on the right area if you do, and in general ships are moving so fast that (laser) shots are not really expected to hit the targeted area reliability. Instead, how about using the 1-5 keys for the turrets to target specific system clusters? 1 for general targeting, 2 for reactors, 3 for engines, and so on. Another option is a single key such as Z to cycle though turrets targeting specific systems which would be displayed on screen. This is not an immediate issue, but freeing up mouse functions to just flying and shooting is always nice.

As an aside, SWS uses keybinds rather than middle click to target ships. T to target the ship directly in front of you, F to cycle though targets, G to target the ship attacking you. I’m having a hard time thinking of advantages this would have compared to the relatively simple mouse controls and the inevitable changes needed since SWS is a team deathmatch game with limited ships, but though I’d put the idea out there. Maybe someone else will have an opinion on it.
![image](https://user-images.githubusercontent.com/36980446/103851638-733a8280-5067-11eb-8653-5328d39162db.png)
Hull or something of the like would be general ship targeting. Advantage of using Z is that it is just one button and cycles through. Advantage of using 1-5 is that you don't have to cycle through them all to get back to the one you want. Plus maybe you can pick more then one to target at the same time, split the targeting between the two.

3.	Target Color

Target coloring is a bit tougher as in SWS there are two distinct sides in a team deathmatch game. Right now since ships are clearly labeled as pirate or civilian, having the pirate ship come up red while civi ships comes up white would be simple. Down the line however, pirates shouldn’t be immediately advertised and there will be different factions either AI or player. At that point it can be as simple as red meaning a ship you are in combat with and green meaning ally or something more complicated with colors to show faction ships such as Faction A ships comes up as orange while Faction B comes up as blue as well as combat/ally colors. Also different structures as different shapes would be nice. Station as a hexagon, landing platform as a square, etc.
![image](https://user-images.githubusercontent.com/36980446/103853730-04135d00-506c-11eb-92da-79bc58344dde.png)

4.	Target Hit Marker

In short, hitting a target with a weapon turns the reticle red or by some other on-screen indicator. Right now the only way you can tell is visually or seeing systems like shields drain. There would need to be a few changes from SWS as where that game’s fighters have (almost) only forward facing guns and thus hitting a ship is always in front and not with turrets, but knowing you are hitting an enemy ship, especially at a distance, would be a nice flight mechanic improvement.
![image](https://user-images.githubusercontent.com/36980446/103850389-58b2da00-5064-11eb-9cee-cf48fb81d800.png)
Better version needed, but still a clear indicator of ships that are hit

**Comment by tsunamayo** [comment]
Okay I havent read it deeply, but I think I agree with all of that! I wanted to try out Squadron too, I guess there are a few interesting thing to learn from them (my guess it is user friendly). 
I think I will drop the middle mouse button to have a toggle between target in front of the reticle logic. The middle mouse select works best if the ship control is avorion like, which is not what we do.

**Comment by ZachZent** [comment]
I would love to fly with you. Or against you :D Empire for life! Squadrons is quite user friendly, easy to learn kind of hard to master

**Comment by tsunamayo** [comment]
Yep why not! I first have to grab the game...

**Comment by tsunamayo** [comment]
@ZachZent Hi, could you send me your graveyard world, could be quite convenient for me to test a few things when I work on that. Cheers

**Comment by ZachZent** [comment]
Enjoy my many ships. 6 months of working on a single iterated engine design which never worked out... still trying. Random ships half done. A few old ones pre-workshop. A number of station modules from the contest. And more!

[aaa.zip](https://github.com/tsunamayo/Starship-EVO/files/5891372/aaa.zip)
